### PR TITLE
Get back to 100% test coverage

### PR DESCRIPTION
**Pull request by johtoblan**
**Summary**:
Changes local types of fileUUID and namespace to str, to check edge case where namespace or fileUUID resolved to \"\". (Cases where metadata_identifier is either ":", "VALIDNS:" or ":VALIDUUID"
Also extended testing of missing namespace
Returns test-coverage to 100%

**Suggested reviewer(s)**:

**Reviewer checklist**:

* [ ] The headers of all files contain a reference to the repository license
* [ ] 100% test coverage of new code - meaning:
  * [ ] The overall test coverage increased or remained the same as before
  * [ ] Every function is accompanied with a test suite
  * [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
  * [ ] Functions with optional arguments have separate tests for all options
* [ ] Examples are supported by doctests
* [ ] All tests are passing
* [ ] All names (e.g., files, classes, functions, variables) are explicit
* [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see [General Conventions](https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html)). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

**Comment by johtoblan** [comment]
this is of course _!!VERY HIGH PRIORITY!!_

File: dmci/api/worker.py [hunk 1]
```diff
@@ -177,8 +177,8 @@ def _extract_metadata_id(self, xml_doc):
         """
         self._file_metadata_id = None
         self._namespace = None
-        fileUUID = None
-        namespace = None
+        fileUUID = ""
```
Comment (mortenwh) [inline dmci/api/worker.py:180]: why this change?

File: tests/test_api/test_worker.py [hunk 2]
```diff
@@ -210,7 +210,13 @@ def testApiWorker_ExtractMetaDataID(filesDir, mockXml):
         '</mmd:mmd>'
     )%("a1ddaf0f-cae0-4a15-9b37-3468e9cb1a2b")
 
-    namespaced_UUID_bad = (
+    namespaced_UUID_empty_return = (
```
Comment (mortenwh) [inline tests/test_api/test_worker.py:213]: variable name seems misleading. It's the missing namespace, which is the problem, isn't it?

**Review by mortenwh — CHANGES_REQUESTED** [review]
I don't understand you coverage comment. We have 100% coverage in master, don't we?

And a few small things as mentioned in the comments. 

Can you also explain the three(?) situations in the test docstring? Maybe we also need a better explanation in _extract_metadata_id. Is it correct that xml_doc is here an extract from the inserted MMD file? And that you initiate the object with the MMD files in the test but override the metadata_identifier by giving the strings defined in the test as input? Is this docstring for _extract_metadata_id better then?

"""
Test the format of the mmd:metadata_identifier to make sure it has both a namespace and a uuid. Returns boolean true or false.

Parameters
----------------
xml_doc : bytecode
    mmd metadata identifier field in bytecode XML format
"""
Something like that?

**Comment by johtoblan** [comment]
https://app.codecov.io/gh/metno/discovery-metadata-catalog-ingestor 99.76% coverage

File: dmci/api/worker.py (hunk 1 above)
Comment (johtoblan) [reply dmci/api/worker.py:180]: len(None) fails, len("") == 0

File: tests/test_api/test_worker.py (hunk 2 above)
Comment (johtoblan) [reply tests/test_api/test_worker.py:213]: ":a1ddaf0f-cae0-4a15-9b37-3468e9cb1a2b".split(":") returns two strings: "" and "a1ddaf0f-cae0-4a15-9b37-3468e9cb1a2b"

Comment (johtoblan) [reply tests/test_api/test_worker.py:213]: This test invokes empty return for namespace.

Comment (mortenwh) [inline tests/test_api/test_worker.py:213]:
```suggestion
    # ":a1ddaf0f-cae0-4a15-9b37-3468e9cb1a2b".split(":") returns two strings: 
    # "" and "a1ddaf0f-cae0-4a15-9b37-3468e9cb1a2b"
    # This test invokes empty return for namespace
    namespaced_UUID_empty_return = (
```

File: tests/test_api/test_worker.py [hunk 3]
```diff
@@ -210,7 +210,13 @@ def testApiWorker_ExtractMetaDataID(filesDir, mockXml):
...
+        '<mmd:mmd xmlns:mmd="http://www.met.no/schema/mmd" xmlns:gml="http://www.opengis.net/gml">'
+        '  <mmd:metadata_identifier>%s</mmd:metadata_identifier>'
+        '</mmd:mmd>'
+    )%(":a1ddaf0f-cae0-4a15-9b37-3468e9cb1a2b")
+
+    namespaced_UUID_too_many = (
```
Comment (mortenwh) [inline tests/test_api/test_worker.py:219]:
```suggestion
    # format "test:no" is wrong (should be "test.no")
    namespaced_UUID_too_many = (
```

**Review by mortenwh — CHANGES_REQUESTED** [review]
Kan være nyttig å få inn et par kommentarer :)

File: tests/test_api/test_worker.py (hunk 3 above)
Comment (johtoblan) [reply tests/test_api/test_worker.py:222]:
```suggestion
    # format "test:no:a1ddaf0f-cae0-4a15-9b37-3468e9cb1a2b" is wrong (should be "test.no:UUID")
    namespaced_UUID_too_many = (
```

**Review by mortenwh — APPROVED** [review]
